### DISCUSSION TITLE
Is there a way to access a map-like object directly instead of by `.get` and `.set`

### DISCUSSION THREAD

**Discussion by NE-SmallTown** [mobxjs/mobx-state-tree#1810] [Q&A]
I have a structure like: 

```
const usersBySchool = {
  'schoolFoo': { name: 'tom', id: '123', ... },
  'schoolBar': { name: 'jerry', id: '789', ... }
}
```

If I define the types like:

```
const UserBySchool = types.mode({
  usersBySchool: types.map(usersBySchool)
})
```

#### when `get`

I **have to** access the instance by `store.usersBySchool.get(school)`, but sometimes I **need to pass the instance to a 3-rd party library** like `<Lib data = { store.usersBySchool } name={ school } />`, in that library, it will access the field by `props.data[props.name]`, so it will throw an error and the tough thing is that I cant modify the 3-rd party library.

#### when `set`

I **have to** set the instance by `const users = store.usersBySchool.get(school); const user = users['xx']; users.set('xxx', { name: 'newName', ...user })`.

If it could be `store.usersBySchool.get(school)['xx'].name = 'newName'`, that would more easier and convenient.

### Expected

So, I want to ask that is there any way to define a key-value object, like the key of the object must be a string, and the value shape of that key must be something like `usersBySchool`, I notice that `types.refinement` maybe can do this but cant figure it out. Thanks!

**Comment by k-g-a** [comment]
As for passing it down to uncontrolled external library - you could pass `getSnapshot(store.usersBySchool)`. 
But if the library within it's own code will modify the data by doing `data['xx'] = newData`, such changes won't propagate back to your model. This should not be the case, if the library follows the DDAU.
As for the `set` part - no, there is no way for `Map` type to do so.
Maybe you should consider using simble mobx's `observable.object()` in a computed or volatile state.
